src/auto-evo/selection_pressure/SelectionPressure.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace AutoEvo;
+
+using System;
+using System.Collections.Generic;
+
+public abstract class SelectionPressure
+{
+    public readonly float Strength;
+    public readonly List<IMutationStrategy<MicrobeSpecies>> Mutations;
+
+    public SelectionPressure(float strength, List<IMutationStrategy<MicrobeSpecies>> mutations)
+    {
+        Strength = strength;
+        Mutations = mutations;
+    }
+
+    public abstract float Score(Species species, SimulationCache cache);
+    public abstract float GetEnergy();
+
+    // TODO: Make sure this does not make things worse
+    public float CacheScore(MicrobeSpecies species, SimulationCache cache)
```

Comment:
Why is there a separate method like this? Wouldn't it be better to bake this into the Score method (and make it non-abstract) and then add a new protected abstract method called something like `CalculateScore`.